documentation/EN/Chapter_9.md

```diff
@@ -0,0 +1,124 @@
+
+_AODVv2_ specifies that control messages have to be mapped to a container called MANET (Generalized Mobile Ad-Hoc Network Packet/Message Format) [RFC5444]. This package format provides a single encapsulation for multiple Ad-Hoc routing protocols.
+
+RFC5444 makes the transmission of control messages more efficient, structuring the content by reducing the number of bytes to transmit.
+
+The RFC5444 format defines the following elements:
+- Package: It is the highest level entity. A package contains a header and zero or more messages.
+- Message: It is the entity that transports the protocol information. A message consists of a header, a TLV (type-length-value) block, and an addresses block.
+- Addresses block: consists of one or more addresses, and an attribute block.
+- TLV Block: It consists in one or more TLVs.
+- TLV: It is a structure (type-length-value) where: 
+ - Type: It is the identifier of the data type that follows.
+ - Length: This field indicates how many bytes the field value occupies.
+ - Value: It is the concrete value of the object to which it refers.
+
+The following image represents the structure of an _RFC544_ package and its dependencies.
+
+ <img src="imple_pic/rfc5444-pkt.png" alt="drawing" height="600" width="1000" align="center"/>
+
+Each type of control message has to be adapted to the format of the _RFC5444_ package.
+
+_AODVv2_ doesn´t require access to the package header [RFC5444].
+
+The _AODVv2_ package header use:
+- **msg-type**. 
+- **msg-hop-limit**.
+- **msg-addr-length**.
+
+**msg-addr-length** indicates the size of the address in the message, whose value corresponds to _addr_length in octets -1_; for example, for _IPV4_ it would be 3 and for _IPV6_ it would be equal to 15.
+
+This is why we will first review which are the header fields of the _RFC5444_ package that _AODVv2_ uses.
+
+
+
+## 9.1 Package info.
+This section of the message allows to quickly know information such as:
+
+- Package version.
+- Sequence number.
+- _Flags_ that allow knowing if the TLVs are in the package without deserializing it.
+
+### 9.1.1 pkt-header
+
+We define it as follows:
+
+**version**: It is a field that contains a 4-bit unsigned integer and specifies the version in which the package and the message content have been built.
+
+**pkt-flag**: It is a 4-bit field that specifies the interpretation of the rest of the package header:
+
+ - **bit 0** (phaseseqnum) if one ('1') is indicated, then it means that the **pkt-seq-num** is included in the **pkt-header**. Otherwise, it is not.
+
+ - **bit 1** (phastlv): If one ('1') is indicated, it means that the **tlv-block** is included in the **pkt-header**. Otherwise, it is not.
+
+ - **bit 2-3**: They are reserved and should be cleaned in each transmission and must be ignored in each reception.
+ 
+**pkt-seq-num**: It is ignored if the _phaseseqnum-flag_ is set to zero ('0'). Otherwise, it is a 16-bit unsigned integer, specifying the sequence number of a package.
```

Comment:
```suggestion
**pkt-seq-num**: It is ignored if the _phaseseqnum-flag_ is set to zero ('0'). Otherwise, it is a 16-bit unsigned integer, specifying the sequence number of a packet.
```

Reply:
hecho 👍